tests/rados/test_pg_split.py

```diff
@@ -79,48 +103,83 @@ def run(ceph_cluster, **kw):
             ):
                 log.error("Failed to delete objects from pool.")
                 raise Exception("Failed to delete objects from pool.")
+            log.info("Completed deletion of objects from the pool")
+
         time.sleep(40)
+        log.debug(
+            f" waiting for PGs to settle down on pool {pool['pool_name']} after applying bulk flag"
+        )
         method_should_succeed(wait_for_clean_pg_sets, rados_obj, timeout)
-        endtime = datetime.datetime.now() + datetime.timedelta(seconds=400)
+
+        endtime = datetime.datetime.now() + datetime.timedelta(seconds=3000)
         while datetime.datetime.now() < endtime:
-            new_prop = rados_obj.get_pool_property(
+            pool_pg_num = rados_obj.get_pool_property(
                 pool=pool["pool_name"], props="pg_num"
-            )
-            if new_prop["pg_num"] > prop["pg_num"]:
+            )["pg_num"]
+            if pool_pg_num == pg_count_bulk_true:
+                log.debug(f"PG count on pool {pool['pool_name']} is achieved")
                 break
-            log.error(
-                f"Actual pg_num {new_prop['pg_num']} is expected to be greater than {prop['pg_num']}"
+            log.debug(
+                f"PG count on pool {pool['pool_name']} has not reached desired levels."
+                f"Expected : {pg_count_bulk_true}, Current : {pool_pg_num}"
             )
-            log.info("Sleeping for 40 secs")
-            time.sleep(40)
+            log.info("Sleeping for 20 secs and checking again")
+            time.sleep(20)
         else:
             raise Exception(
-                f"Actual pg_num {new_prop['pg_num']} is expected to be greater than {prop['pg_num']}"
+                f"pg_num on pool {pool['pool_name']} did not rach the desired levels of PG count with bulk flag enabled"
+                f"Expected : {pg_count_bulk_true}"
+            )
+        log.info(
+            "PGs increased to desired levels after application of bulk flag on the pool"
+        )
+        time.sleep(60)
+
+        log.info(
+            f"Proceeding to remove the bulk flag from the pool : {pool['pool_name']}"
+        )
+        if not pool_obj.rm_bulk_flag(pool["pool_name"]):
+            log.error(
+                f"Could not remove the bulk flag from the pool {pool['pool_name']}"
             )
-        pool_obj.rm_bulk_flag(pool["pool_name"])
-        time.sleep(10)
+            raise Exception("Expected bulk flag should be False.")
+        time.sleep(30)
```

Comment:
If 30 sec blind sleep in required for bulk flag to become false after setting it to false, then this sleep should move above the IF condition check ` if not pool_obj.rm_bulk_flag(pool["pool_name"]):`